helix-lsp/src/client.rs

```diff
@@ -1022,21 +1035,10 @@ impl Client {
 
     pub fn resolve_completion_item(
         &self,
-        completion_item: lsp::CompletionItem,
-    ) -> Option<impl Future<Output = Result<lsp::CompletionItem>>> {
-        let capabilities = self.capabilities.get().unwrap();
-
-        // Return early if the server does not support resolving completion items.
-        match capabilities.completion_provider {
-            Some(lsp::CompletionOptions {
-                resolve_provider: Some(true),
-                ..
-            }) => (),
-            _ => return None,
-        }
-
-        let res = self.call::<lsp::request::ResolveCompletionItem>(completion_item);
-        Some(async move { Ok(serde_json::from_value(res.await?)?) })
+        completion_item: &lsp::CompletionItem,
```

Comment:
Ah nice, I remember thinking there was an unnecessary clone in completion resolving because of the type here lacking a ref :+1: